src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -173,8 +173,7 @@ public static function get_core_data() {
 		 * @param WP_Theme_JSON_Data $theme_json Class to access and update the underlying data.
 		 */
 		$theme_json   = apply_filters( 'wp_theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
-		$config       = $theme_json->get_data();
-		static::$core = new WP_Theme_JSON( $config, 'default' );
```

Comment:
What's with the different `$origin` here (`default` instead of `theme`)?  Is that still taken into account and do we have tests covering that?

Reply:
So there are 4 `$origins`, `default` (json coming from the core), `theme` (coming from theme), `block` (coming from blocks, mostly plugins), `custom` (custom user entries). 

Since this function is related to core_data, we use `default` here. Same is with other functions below.

Reply:
This PR changes the origin everywhere, unless I‘m mistaken. Hence the question.

Reply:
@swissspidy My bad, I misunderstood the question.

This PR doesn't affect the origin. If you see the filter just one line above we are creating a new object for `WP_Theme_JSON_Data` with the same origin `new WP_Theme_JSON_Data( $config, 'default' )`. 

Inside the [WP_Theme_JSON_Data::_construct](https://github.com/WordPress/wordpress-develop/blob/002c2cc52623f8b1fb0aca20a7833ff969c3b19c/src/wp-includes/class-wp-theme-json-data.php#L44), you will find it simply sets a private property `theme_json` ( `$this->theme_json = new WP_Theme_JSON( $data, $this->origin );` ). In this PR we are only trying to access the object set in `private $theme_json` directly.

This way we can avoid, `object -> to JSON -> back to the same object`.


**Special Case** 
If the user decides to use the filter to create a new object with a different origin; In that case, I feel that it would be the developer's fault for using the wrong origin. In most of the existing popular plugins I have seen they are using the right $origin in each filter.